### PR TITLE
Changed popup maximum dimensions from absolute units

### PR DESCRIPTION
**Pull request by detly**
Currently the popups are wider than my (portrait mode) screen. The simplest fix I could see was to make it proportional to the view dimensions instead. This might have some drawbacks, but I'm happy to take suggestions.

### CODE REVIEW

**Comment by niosus** [comment]
Can you summarize shortly why there are so many changes? You've touched quite a lot of code here.

**Comment by niosus** [comment]
Also, please ignore the build failures for now, I guess it is because of this: https://lintlyci.github.io/Flake8Rules/rules/W605.html 

Hopefully will fix it today.

**Comment by detly** [comment]
Ugh, even though I rebased on the dev branch I think I messed up the PR.
There are not actually that many changes. I will close and redo.

**Comment by niosus** [comment]
Thanks @detly
